provision/opentofu/modules/rosa/hcp/variables.tf

```diff
@@ -0,0 +1,82 @@
+variable "token" {
+  type      = string
+  sensitive = true
+}
+
+variable "region" {
+  description = <<-EOT
+    Region to create AWS infrastructure resources for a
+      ROSA with hosted control planes cluster. (required)
+  EOT
+  type        = string
+}
+
+variable "path" {
+  description = "(Optional) The arn path for the account/operator roles as well as their policies."
+  type        = string
+  default     = null
+}
+
+variable "cluster_name" {
+  description = "Name of the ROSA hosted control planes cluster to be created."
+  type        = string
+  default     = "rosa-hcp"
+
+  validation {
+    condition     = can(regex("^[a-z][-a-z0-9]{0,13}[a-z0-9]$", var.cluster_name))
+    error_message = <<-EOT
+      ROSA cluster names must be less than 16 characters.
+        May only contain lower case, alphanumeric, or hyphens characters.
+    EOT
+  }
+}
+
+variable "vpc_cidr" {
+  description = <<-EOT
+    IPv4 CIDR netmask for the VPC resource.
+      This should equal or include the cluster's Machine CIDR netmask.
+  EOT
+  type        = string
+  default     = "10.0.0.0/16"
+
+  validation {
+    condition     = can(cidrnetmask(var.vpc_cidr))
+    error_message = "VPC CIDR must be a valid CIDR netmask, e.g., '10.0.0.0/16'."
+  }
+}
+
+variable "subnet_cidr_prefix" {
+  description = <<-EOT
+    The CIDR prefix value to use when dividing up the VPC CIDR range into subnet ranges.
+      E.g., 24 to create equal subnets of size "24": 10.0.1.0/24, 10.0.2.0/24, etc.
+  EOT
+  type        = number
+  default     = 24
+}
+
+variable "private_subnets_only" {
+  description = "Only create private subnets"
+  type        = bool
+  default     = false
+}
+
+variable "extra_tags" {
+  description = "Extra tags to apply to AWS resources"
+  type        = map
+  default     = {}
+}
+
+variable "openshift_version" {
+  type    = string
+  default = "4.14.5"
+}
+
+variable "instance_type" {
+  type    = string
+  default = "m5.4xlarge"
+}
+
+variable "replicas" {
+  type    = number
+  default = 2
+}
```

Comment:
Nitpick: As we now have a good place here for the defaults, I would tend to not keep them in other places and just maintain them here. 

So we would have fewer parameters on our workflows. It would also be ok to remove some and keep some from the list if you think we're about to use them now and then.

As this is a nitpick, feel free to keep for another future PR.

Reply:
I've removed the default values from the GH rosa-cluster-create workflow, but left the inputs so that the default can still be overridden. The `rosa_create_cluster.sh` script then adds the variables as `-var ` args to `tofu` if a value is specified.